Docs/developer_guide/module_overview.md

```diff
@@ -0,0 +1,160 @@
+# Module Overview
+
+Slicer supports multiple types of modules:
+* [Command Line Interface (CLI)](#command-line-interface-cli)
+* [Loadable](#loadable-modules)
+* [Scripted](#scripted-modules).
+
+While the developer has to choose between one of the types to implement its module, the end user will NOT notice the difference as they all share the same look & feel.
+
+The choice for a given type of module is usually based on the type of inputs/parameters for a given module.
+
+The [Extension Wizard](user_guide/modules/extensionwizard.md#extension-wizard) module provides a graphical interface within Slicer to aid in the creation of Slicer extensions and associate modules.
+
+## Command Line Interface (CLI)
+
+CLIs are standalone executables with a limited input/output arguments complexity (simple argument types, no user interactions...). They are typically implemented in C++ (e.g using [ITK](https://itk.org)).
+
+Slicer allows running (almost) any Python script as a CLI module by providing an interface description .xml file.
+
+Specifications:
+* Shared library, executable, or Python script.
+* UI automatically generated.
+
+Getting started:
+* Create initial skeleton using the [Extension Wizard](user_guide/modules/extensionwizard.md#extension-wizard).
+* Learn from the existing Slicer [CLI modules](https://github.com/Slicer/Slicer/tree/main/Modules/CLI).
+
+More information:
+* [Developing and contributing extensions for 3D Slicer](https://www.slicer.org/wiki/Documentation/Nightly/Training#Developing_and_contributing_extensions_for_3D_Slicer)
+* [Tutorials for software developers](https://www.slicer.org/wiki/Documentation/Nightly/Training#Tutorials_for_software_developers)
+* [Slicer execution model](https://www.slicer.org/wiki/Documentation/Nightly/Developers/SlicerExecutionModel)
+* [CLI in Slicer4](http://www.na-mic.org/Wiki/index.php/File:Slicer4_CLI.ppt) (presentation of 2012 NAMIC AHM)
```

Comment:
Although even `Tutorials for software developers` page should eventually be moved out of the wiki and into the sphinx docs, right? Or does the wiki still serve a purpose currently?

Reply:
>should eventually be moved out of the wiki and into the sphinx docs, right?

Indeed, this is now formally tracked in https://github.com/Slicer/Slicer/issues/6676

> does the wiki still serve a purpose currently?

Since there are still documentation that need to be migrated, it is the authoritative documentation for some specific components.